src/StepperFirmata2.cpp

```diff
@@ -0,0 +1,401 @@
+  Copyright (C) 2010-2011 Paul Stoffregen.  All rights reserved.
+  Copyright (C) 2009 Shigeru Kobayashi.  All rights reserved.
+  Copyright (C) 2013 Norbert Truchsess. All rights reserved.
+  Copyright (C) 2009-2016 Jeff Hoefs.  All rights reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  See file LICENSE.txt for further informations on licensing terms.
+
+  Last updated by Jeff Hoefs: January 23rd, 2016
+*/
+
+#include <ConfigurableFirmata.h>
+#include "StepperFirmata2.h"
+#include "utility/AccelStepper.h"
+#include "utility/MultiStepper.h"
+
+boolean StepperFirmata2::handlePinMode(byte pin, int mode)
+{
+  if (mode == PIN_MODE_STEPPER) {
+    if (IS_PIN_DIGITAL(pin)) {
+      digitalWrite(PIN_TO_DIGITAL(pin), LOW); // disable PWM
+      pinMode(PIN_TO_DIGITAL(pin), OUTPUT);
+      return true;
+    }
+  }
+  return false;
+}
+
+void StepperFirmata2::handleCapability(byte pin)
+{
+  if (IS_PIN_DIGITAL(pin)) {
+    Firmata.write(PIN_MODE_STEPPER);
+    Firmata.write(21); //21 bits used for number of steps
+  }
+}
+
+void StepperFirmata2::reportPosition(byte deviceNum, bool complete)
+{
+  if (stepper[deviceNum]) {
+    
+    byte data[5];
+    long position = stepper[deviceNum]->currentPosition();
+    encode32BitSignedInteger(position, data);
+    
+    Firmata.write(START_SYSEX);
+    Firmata.write(ACCELSTEPPER_DATA);
+    if (complete) {
+      Firmata.write(STEPPER_MOVE_COMPLETE);
+    } else {
+      Firmata.write(STEPPER_REPORT_POSITION);
+    }
+    Firmata.write(deviceNum);
+    Firmata.write(data[0]);
+    Firmata.write(data[1]);
+    Firmata.write(data[2]);
+    Firmata.write(data[3]);
+    Firmata.write(data[4]);
+    Firmata.write(END_SYSEX);
+  }
+}
+
+void StepperFirmata2::reportGroupComplete(byte deviceNum)
+{
+  if (group[deviceNum]) {
+    
+    Firmata.write(START_SYSEX);
+    Firmata.write(ACCELSTEPPER_DATA);
+    Firmata.write(MULTISTEPPER_MOVE_COMPLETE);
+    Firmata.write(deviceNum);
+    Firmata.write(END_SYSEX);
+  }
+}
+
+/*==============================================================================
+ * SYSEX-BASED commands
+ *============================================================================*/
+
+boolean StepperFirmata2::handleSysex(byte command, byte argc, byte *argv)
+{
+    
+  if (command == ACCELSTEPPER_DATA) {
+
+    byte stepCommand, deviceNum, directionPin, stepPin, stepDirection, groupNum;
+    byte interface, wireCount, stepType;
+    byte motorPin3, motorPin4, enablePin, invertPins = 0;
+    unsigned int minPulseWidth;
+    long numSteps;
+    int maxSpeed, stepSpeed, accel, decel;
+
+    unsigned int index = 0;
+
+    stepCommand = argv[index++];
+    deviceNum = argv[index++];
+  
+    if (deviceNum < MAX_STEPPERS) {
+
+      if (stepCommand == STEPPER_CONFIG) {
```

Comment:
I guess for a first time user, hearing acceleration is probably more satisfying than a standard step sequence without acceleration, but in either case it may be helpful to have a default.